### DISCUSSION TITLE
NBT-EVO ssh access and AA Headunit

### DISCUSSION THREAD

**Discussion by alirana01** [BimmerGestalt/AAIdrive#828] [Ideas]
Hi,

I was going over some forums and found out that it is possible to have access to NBT-EVO via ssh. The actual tutorial I believe can be found out with a google search it was to change the car style and clock styles. Anyway my point is since we have ssh root level access possible. would it be possible somehow that the desktop-headunit emulator from android sdk be loaded on the nbt evo and we can launch it via some script. Not very expert on how the iDrive and the headunit software is but with some help and support I maybe able to work this out. I have the ENET cable on order so once that arrives this is going to be an interesting thing to try. Your ideas on this will be very helpful

**Comment by hufman** [comment]
I suppose, in the realm of software, anything is possible!
Such a project would involve compiling the Headunit Emulator for QNX ARM, targeting it to whatever unique graphics system the car uses, and then hooking it into the proper menuing and networking systems.
I would have no idea how to get started. Maybe one of the AA dongle companies might have the skill, but the market for people who have rooted their car will be very tiny, and the engineering effort would be incredibly expensive and impossible to earn back.

**Reply by alirana01** [reply]
Yeah you are right about that. Okay lets think of something else, I am sure you may have gone through a lot of this already but I just wanted to play around a bit more with this using your code. TBH I still have to explore the repo to find more details. Anyway lets consider we have the AA headunit emulator compiled for this. Is there a way to move that to the car using your APIs and then trying to launch a process on car with the phone connected to the car's usb to launch android auto. If there is a way we need to have the android auto server running on the phone because that's how the emulator communicates with the phone.

Just a lot of speculation here to a far far fetched idea but only your repo is well maintained to ask or discuss this stuff.

Thanks for patience in my wild and probably not so feasible goose chase of an idea

**Reply by hufman** [reply]
As you explore the code and documentation, you'll find that no code or process is being run in the car. Instead, a widget layout is sent to the car, and then the contents of the widgets are updated over time in response to button events from the car.
The only option, using the protocol that AAIdrive uses to connect with the car, is for the phone to run an Android Auto headunit, screenshot it, and send each screenshot to the car.  This has the poor UX of needing to enable Developer Mode in AA at each connection, and the terrible performance of shipping fullscreen JPGs over Bluetooth to the car (as seen in AAIdrive's Map and the Screen Mirroring addon), and there is no touchscreen support in the protocol.
A much better solution for the Android Auto experience is to use the dedicated AA dongles, which seem to work very well.
